django/core/validators.py

```diff
@@ -163,29 +163,27 @@ def validate_integer(value):
 
 @deconstructible
 class EmailValidator:
+    # Matches https://html.spec.whatwg.org/multipage/forms.html#valid-e-mail-address
     message = _('Enter a valid email address.')
     code = 'invalid'
     user_regex = _lazy_re_compile(
-        r"(^[-!#$%&'*+/=?^_`{}|~0-9A-Z]+(\.[-!#$%&'*+/=?^_`{}|~0-9A-Z]+)*\Z"  # dot-atom
-        r'|^"([\001-\010\013\014\016-\037!#-\[\]-\177]|\\[\001-\011\013\014\016-\177])*"\Z)',  # quoted-string
+        r"^[a-z0-9.!#$%&'*+\/=?^_`{|}~-]+\Z",
         re.IGNORECASE)
     domain_regex = _lazy_re_compile(
-        # max length for domain name labels is 63 characters per RFC 1034
-        r'((?:[A-Z0-9](?:[A-Z0-9-]{0,61}[A-Z0-9])?\.)+)(?:[A-Z0-9-]{2,63}(?<!-))\Z',
+        r'[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?(?:\.[a-z0-9](?:[a-z0-9-]{0,61}[a-z0-9])?)*\Z',
         re.IGNORECASE)
     literal_regex = _lazy_re_compile(
         # literal form, ipv4 or ipv6 address (SMTP 4.1.3)
         r'\[([A-f0-9:\.]+)\]\Z',
         re.IGNORECASE)
-    domain_whitelist = ['localhost']
 
     def __init__(self, message=None, code=None, whitelist=None):
         if message is not None:
             self.message = message
         if code is not None:
             self.code = code
         if whitelist is not None:
-            self.domain_whitelist = whitelist
+            pass  # deprecated
```

Comment:
What's implemented right now isn't a deprecation -- it's a backwards-incompatible change since `whitelist` is completely ignored. It might be better to simply remove the parameter so that projects using it will be loudly notified about the change. @collinanderson what was your thinking here?